src/iface/ethernet.rs

```diff
@@ -734,6 +944,61 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
         }
     }
 
+    /// Host duties of the **IGMPv2** protocol
+    ///
+    /// Sets up `igmp_report_state` for responding to IGMP
+    /// general/specific membership queries.
+    #[cfg(all(feature = "proto-ipv4", feature = "proto-igmp"))]
+    fn process_igmp<'frame>(&mut self, timestamp: Instant, ipv4_repr: Ipv4Repr, ip_payload: &'frame [u8]) ->
+                             Result<Packet<'frame>> {
+        let igmp_packet = IgmpPacket::new_checked(ip_payload)?;
+        let igmp_repr = IgmpRepr::parse(&igmp_packet)?;
+
+        // for now - reply immediately
+        match igmp_repr {
+            IgmpRepr::MembershipQuery { group_addr, version, max_resp_duration } => {
+                // General Query
+                if group_addr.is_unspecified() && ipv4_repr.dst_addr == Ipv4Address::MCAST_ALL_SYSTEMS {
+                    if let Some((first_group_addr, &())) = self.ipv4_mcast_groups.iter().next() {
+                        let interval = match version {
+                            IgmpVersion::Version1 =>
+                                Duration::from_millis(100),
+                            IgmpVersion::Version2 => {
+                                // No dependence on a random generator
+                                // but at least spread reports evenly
+                                // over max_resp_time
+                                let intervals = self.ipv4_mcast_groups.len() as u32 + 1;
+                                max_resp_duration / intervals
+                            }
+                        };
+                        self.igmp_report_state = IgmpReportState::ToGeneralQuery(
+                            version,
+                            timestamp + interval,
+                            interval,
+                            first_group_addr.clone()
+                        );
```

Comment:
Can you explain why you cannot just report immediately? Preferably in the comments.